client/my-sites/email/titan-setup-mailbox/index.js

```diff
@@ -60,22 +64,21 @@ const TitanSetupMailbox = ( { selectedDomainName } ) => {
 
 	return (
 		<>
-			<PageViewTracker
-				path={ analyticsPath }
-				title="Email Management > Set up your Professional Email"
-			/>
+			<PageViewTracker path={ analyticsPath } title="Email Management > Set Up Titan Mailbox" />
 
 			{ selectedSite && <QuerySiteDomains siteId={ selectedSite.ID } /> }
 
 			<Main wideLayout={ true }>
-				<DocumentHead title={ translate( 'Set up your Professional Email' ) } />
+				<DocumentHead title={ translate( 'Set Up Mailbox' ) } />
 
 				<EmailHeader currentRoute={ currentRoute } selectedSite={ selectedSite } />
 
-				<HeaderCake onClick={ goToCustomerHome }>
-					{ translate( 'Set up your Professional Email' ) }
+				<HeaderCake onClick={ handleBack }>
+					{ getTitanProductName() + ': ' + selectedDomainName }
 				</HeaderCake>
 
+				<SectionHeader label={ translate( 'Set Up Mailbox' ) } />
```

Comment:
I think it makes sense to have this be capitalised, but I am not sure if we should handle that via CSS or via separate strings. (We now have both `Set up mailbox` and `Set Up Mailbox` as translated strings.)